src/renderers/gl/renderer.cpp

```diff
@@ -217,6 +217,9 @@ void mrg::Renderer::render(mg::RenderableList const& renderables) const
     // Deleting unused textures only requires the GL context. This clean-up
     // does not affect screen contents so can happen after swap_buffers...
     texture_cache->drop_unused();
+
+    if (auto gl_error = glGetError())
+        mir::log_debug("GL error: %d", gl_error);
```

Comment:
Useful? We can enable printing of all GL errors with MESA_DEBUG=1.

Also glGetError fetches the first in a queue of errors, see https://www.khronos.org/opengl/wiki/OpenGL_Error
which means an error printed here may not be related to the last gl function call.

It doesn't hurt though.

Reply:
I know it is of limited use, but at least *something* gets logged to indicate an error.

Would you suggest this?

   while (auto gl_error = glGetError())
       mir::log_debug("GL error: %d", gl_error);

I'm happy to remove it though, if you feel that would be better.

Reply:
I'm ok with keeping it in, to at least indicate an error happened.